### PR TITLE
Initial approach on posterhall performance optimisation

### PR DESCRIPTION
**Pull request by mike-lvov**
Given that the amount of posters in `posterhall` is expected to be > than 1600, we need a way to display the thing in an efficient & performant way

### CODE REVIEW

File: src/hooks/useDebounceSearch.ts [hunk 1]
```diff
@@ -15,9 +15,13 @@ export const useDebounceSearch = () => {
...
     () =>
-      debounce((value: string) => {
-        setSearchQuery(value.toLowerCase());
-      }, SEARCH_DEBOUNCE_TIME),
+      debounce(
+        (value: string) => {
+          setSearchQuery(value.toLowerCase());
+        },
+        SEARCH_DEBOUNCE_TIME,
+        { maxWait: SEARCH_DEBOUNCE_TIME * 3 }
```
Comment (mike-lvov) [inline src/hooks/useDebounceSearch.ts:23]: For some reason, the execution of the denounced func was delayed for unreasonable amount of time (>5 seconds)

**Comment by mike-lvov** [comment]
@0xdevalias merging this just to test the changes with real-world data. Please, make a retrospective review of this

**Comment by 0xdevalias** [comment]
@mike-lvov Please make sure to add these 'standalone PRs' (eg. where they aren't linked to an issue) directly to the Sprint project board so that we have an accurate reference of the work done within the sprint.

File: src/hooks/useDebounceSearch.ts (hunk 1 above)
Comment (0xdevalias) [reply src/hooks/useDebounceSearch.ts:23]: Was this while experiencing other performance issues/where the browser was already struggling? If so, might have been related to that more than our use of `debounce` here?

Comment (0xdevalias) [reply src/hooks/useDebounceSearch.ts:23]: If we keep this, might be worth extracting `SEARCH_DEBOUNCE_TIME * 3` as a `const` in settings as well.

File: src/hooks/posters.ts [hunk 2]
```diff
@@ -47,37 +49,51 @@ export const usePosters = (posterHallId: string) => {
...
+  const [displayedPostersCount, setDisplayedPostersAmount] = useState(
+    DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT
+  );
+
+  const increaseDisplayedPosterCount = useCallback(() => {
+    setDisplayedPostersAmount(
+      (prevPostersNumber) =>
+        prevPostersNumber + DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT
+    );
+  }, []);
```
Comment (0xdevalias) [inline src/hooks/posters.ts:61]: Would this be better named something like `showMorePosters` or similar?

```typescript
  const showMorePosters = useCallback(() => {
    setDisplayedPostersAmount(
      (prevCount) =>
        prevCount + DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT
    );
  }, []);
```

File: src/hooks/posters.ts [hunk 3]
```diff
@@ -47,37 +49,51 @@ export const usePosters = (posterHallId: string) => {
...
 
   const fuseVenues = useMemo(
     () =>
-      new Fuse(posterVenues, {
+      new Fuse(filteredPosterVenues, {
         keys: ["poster.title", "poster.authorName", "poster.categories"],
       }),
-    [posterVenues]
+    [filteredPosterVenues]
   );
```
Comment (0xdevalias) [inline src/hooks/posters.ts:77]: Why did we move `filteredPosterVenues` up here + make `fuseVenues` consume it?

File: src/hooks/posters.ts [hunk 4]
```diff
@@ -47,37 +49,51 @@ export const usePosters = (posterHallId: string) => {
...
 
   const searchedPosterVenues = useMemo(() => {
-    if (!searchQuery) return posterVenues;
+    if (!searchQuery)
+      return filteredPosterVenues.slice(0, displayedPostersCount);
 
     return fuseVenues
       .search(searchQuery)
+      .slice(0, displayedPostersCount)
```
Comment (0xdevalias) [inline src/hooks/posters.ts:85]: I can see what you're doing, using `.slice` to reduce the number of displayed venues, but I expect the crux of the performance issue here is either in the initial loading of the data, or in processing the parts above. I would be very surprised if it was actually at the 'rendering' part after here that everything slowed down.

**Review by 0xdevalias — COMMENTED** [review]
[Retroactive Review] Retroactive review comments.

**Pull request merged**